src/Confluent.Kafka.Avro/AvroSerdeProviderConfig.cs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016-2018 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System.Collections.Generic;
+using Confluent.SchemaRegistry;
+
+
+namespace Confluent.Kafka.Serialization
+{
+    /// <summary>
+    ///     AvroSerdeProvider configuration properties.
+    /// </summary>
+    public class AvroSerdeProviderConfig : SchemaRegistryConfig
+    {
+        /// <summary>
+        ///     Configuration property names specific the avro serde provider.
+        /// </summary>
+        public new static class PropertyNames
+        {
+            /// <summary>
+            ///     Specifies the initial size (in bytes) of the buffer used for Avro message
+            ///     serialization. Use a value high enough to avoid resizing the buffer, but
+            ///     small enough to avoid excessive memory use. Inspect the size of the byte
+            ///     array returned by the Serialize method to estimate an appropriate value.
+            ///     Note: each call to serialize creates a new buffer.
+            ///
+            ///     default: 1024
```

Comment:
yep. now tracking this in the 1.0 todo list.

Reply:
also related to this are changes under consideration for the serializer api in which the .net client may do it's own memory management rather than relying on the GC.